### ISSUE TITLE
detach_vertex_named down method does not restore orphaned vertices

### ISSUE THREAD

**Issue by baron1405** [crystal-lang/crystal-molinillo#11]
The `detach_vertex_named.up` method removes the specified vertex, any edges pointing to it, outgoing edges pointing from it, and any vertices that would be orphaned after the removal (if they are not root vertices). However, undoing this action by calling the `down` method restores the vertex and edges but does not restore any orphaned vertices. The original Molinillo code behaves the same way. Is this a design decision or a bug?

**Comment by straight-shoota** [comment]
I'm not familiar enought witht the implementation, so I can't really tell. But if the original implementation is the same, I would suppose it's a design decision. At least for this port, the decision is to copy the original behaviour 😏

**Comment by baron1405** [comment]
Thanks for the quick reply. I will open an issue on the Molinillo project to see if they can shed some light on this for us. What concerns me is that no only is the behavior asymmetric and inconsistent with the other actions, it restores dangling edges to the graph. It appears that the `up` method does not clear the edges on the detached vertex itself. When that vertex is restored in the `down` method, it still has outgoing edges pointing to the orphaned vertices.

**Comment by baron1405** [comment]
BTW do you know the status of the Molinillo project? I have created a few issue with no comments, they have not done any updates in a few years, there are open issues and PRs dating back about 3 years.

**Comment by baron1405** [comment]
I figured it out. The detach_vertex_named up method recursively calls detach_vertex_named but through the dependency graph. This pushes more actions on the log. This makes unit testing the detach_vertex_named down method a bit tricky. In general, I am not a fan of the high amount of cyclic coupling between the dependency graph, log, and actions. Closing this issue because is it working as designed and will not leave any inconsistencies. Thanks for working through this with me.
